### ISSUE TITLE
RFE | Use `libcontainers/cgroup` as the way to handle cgroups on kata-containers

### ISSUE THREAD

**Issue by fidencio** [kata-containers/kata-containers#2993]
On the last CRI-O community meeting @mrunalp raised the point that we should consider using `libcontainers/cgroups` as the way to handle cgroups on kata-containers, and the main reason for that being kubernetes defaulting to the `systemd` cgroup driver from its already released 1.22 release (see: https://github.com/kubernetes/kubeadm/issues/2376).

Based on @mrunalp's feedback, I started a thread in the #kata-dev Slack channel, and there @snir911 mentioned that `libcontainers/cgroups` wouldn't properly work with kata-containers, since runc `v1.0.0-rc90/v1.0.0-rc1`.

Let's use this issue here to:
- [ ] evaluate the runc issue that wouldn't allow us to use `libcontainers/cgroups`
- [ ] properly report the issue, so we can then bother @kolyshkin for some help
- [ ] do the switch from the kata-containers side

There's an already going on discussion on systemd cgroup driver being broken with current 2.2+ kata-containers release and this discussion, although similar, is orthogonal to the work done by @snir911 and @gkurz on hot-fixing that one.

**Comment by fidencio** [comment]
@sameo mentioned that we really should aim for having `virtcontainers` directly depending on one, and solely one cgroup package.

**Comment by egernst** [comment]
Why not fix systemd handling  within `containerd/cgroups` ?

**Comment by snir911** [comment]
**Addtional information about the current issue with libcontainer/cgroups**
The issue appeared specifically since https://github.com/opencontainers/runc/commit/b810da149008f1d7d07f481970d40c0d035958af
AFAIU the reason is due to a change in the process of setting a cgroup: `We have to freeze the container while systemd sets the cgroup settings` as stated, this is causing the shim process to freeze itself as that's the process we are setting cgroups for (and not another container process as runc does).

>Why not fix systemd handling within `containerd/cgroups` ?

I agree it would best for us, however, from what I've seen my feeling is that having containerd/cgroups to support what we need will require a quite intrusive change (in api) which is unlikely to be needed by containerd, that's me speculating though, I'm in favor of having more proper investigation / discussion in containerd.

**Comment by egernst** [comment]
Yeah, I'm interested in at least exploring the fix on containerd side. Happy to help.
